### DISCUSSION TITLE
how to achieve the getInitialAppProps functionality in next 14?

### DISCUSSION THREAD

**Discussion by sandeepbfnp** [vercel/next.js#60952] [App Router]
my earlier project is based using getInitialAppProps in_app.js for server and client page rendering both as we have dynamic data to render in server as well. I need a alternative to acheive this in next 14 so i can acheive the same.

**Comment by ImanAkrami** [comment]
In your root `layout.tsx`, which is preferably a server component, make initialize the data (fetching etc). you can either pass it down using context or fetching again in the lower levels ( if it's cached)
